draft-ietf-jsonpath-base.md

```diff
@@ -289,23 +293,25 @@ JSONPath expressions are applied to a JSON value, the *argument*.
 Within the JSONPath expression, the abstract name `$` is used to refer
 to the *root node* of the argument, i.e., to the argument as a whole.
 
-JSONPath expressions can use the *dot notation*
+JSONPath expressions primarily use the *bracket notation*
 
 ~~~~
-$.store.book[0].title
+$['store']['book'][0]['title']
 ~~~~
 
-or the more general *bracket notation*
+but may also use the shorthand *dot notation*
 
 ~~~~
-$['store']['book'][0]['title']
+$.store.book[0].title
 ~~~~
 
 to build paths that are input to a JSONPath implementation.
+Generally, the dot notation is preferred for most applications.
+Mixing these syntaxes within a single path is also useful.
 
 JSONPath allows the wildcard symbol `*` to select any member of an
 object or any element of an array ({{wildcard}}).
-The descendant operators (which start with `..`) select some or all of the descendants ({{descendant-selectors}}) of a node.
+The descendant selector (which starts with `..`) recursively selects some or all of the descendants ({{descendant-selectors}}) of a node.
```

Comment:
Enumerating the descendants is definitely done recursively.  This is how we get descendants of descendants until there are no more descendants.  An implementation doesn't need to process these recursively, but it _is_ recursive by its very nature.

> but once we are dealing with the descendants, it's just a list of nodes and no further recursion need be involved.

I don't understand how this is relevant.  This is after the recursive definition applies.

Reply:
The phrase "select some or all of the descendants" is sufficient without needing to talk about recursively selecting.

Reply:
It's only sufficient if we define "descendants" to mean all of them down the tree (which apparently we have, and recursively)

(line 222)
```
Children (of a node):
: If the node is an array, each of its elements,
  or if the node is an object, each of its member values (but not its
  member names). If the node is neither an array nor an object, it has no children.

Descendants (of a node):
: The children of the node, together with the children of its children, and so forth
  recursively. More formally, the descendants relation between nodes is the transitive
  closure of the children relation.
```

I can change this here since "descendants" is defined as such.